wallet-new/docs/RelatingWalletSpecToCardano.md

```diff
@@ -1520,6 +1520,14 @@ Note: the above code was added shortly after this document was pinned to a git h
 
 Cardano does not implement transaction TTL yet and instead relies on submission count and a maximum retry limit.
 
+Currently, a transaction TTL may be estimated based on the creation timestamp of the transaction.
+The process has a constant number of submission attempts per transaction as well as a constant submission attempt rate.
+We attempt to run the submission function every five seconds, and the maximum attempt count is 255.
+This indicates a TTL of `255 * 5`, which is 1,275 seconds (or 21 minutes and 15 seconds).
```

Comment:
N.B. This might be the baseline, but it's not what happens in reality. If you see how the data layer is configured, we initialise the submission layer like this:

```
submission <- newMVar (newWalletSubmission rho)
...
rho = defaultResubmitFunction (exponentialBackoff 255 1.25)
```

Which means that yes, we try up to a max of 255 times, but crucially we don't reschedule the transaction in the same slot! If you check the implementation of `exponentialBackoff`:

```
-- | An exponential backoff policy, parametric over a maximum number of
-- 'MaxRetries' and an 'Exponent' for the backoff.
exponentialBackoff :: MaxRetries -> Exponent -> RetryPolicy
exponentialBackoff maxRetries exponent submissionCount currentSlot =
    let (delta :: Word) = fromIntegral ((floor (exponent ^^ (getSubmissionCount submissionCount))) :: Int)
    in  limited maxRetries (mapSlot ((+) delta)) submissionCount currentSlot
```

You will see we call `mapSlot ((+) delta))` which means that we are progressively increasing the slot this transaction will be attempted more and more as the retries goes by. Your description would have been correct if we used a `constantRetry` policy.

Sorry for not spotting this earlier!

Reply:
I'm getting something like 4 quintillion years of TTL based on this. Can you check my calculation in the IOHK slack?